pkg/attestation/crafter/materials/oci_image.go

```diff
@@ -281,3 +293,188 @@ func (i *OCIImageCrafter) isLatestTag(ref name.Reference, currentDigest string)
 	i.logger.Debug().Str("name", latestRef.String()).Msg("image does not have a 'latest' tag")
 	return false
 }
+
+// parseLayoutReference parses a layout reference that may include a digest selector.
+func parseLayoutReference(ref string) (string, string) {
+	// Check for @digest suffix
+	if idx := strings.LastIndex(ref, "@"); idx != -1 {
+		return ref[:idx], ref[idx+1:]
+	}
+	return ref, ""
+}
+
+// isOCILayoutPath checks if the given path is a valid OCI layout directory.
+func (i *OCIImageCrafter) isOCILayoutPath(path string) bool {
+	// Check if path exists and is a directory
+	info, err := os.Stat(path)
+	if err != nil || !info.IsDir() {
+		return false
+	}
+
+	// Check for oci-layout file
+	layoutFile := filepath.Join(path, "oci-layout")
+	if _, err := os.Stat(layoutFile); err != nil {
+		return false
+	}
+
+	return true
+}
+
+// craftFromLayout creates a material from an OCI layout directory.
+// If digestSelector is provided, it will look for that specific digest in the layout.
+// Otherwise, it uses the first manifest in the index.
+func (i *OCIImageCrafter) craftFromLayout(_ context.Context, layoutPath, digestSelector string) (*api.Attestation_Material, error) {
+	// Read the OCI layout
+	layoutPath, err := filepath.Abs(layoutPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get absolute path: %w", err)
+	}
+
+	path, err := layout.FromPath(layoutPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read OCI layout: %w", err)
+	}
+
+	// Get the image index
+	index, err := path.ImageIndex()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read image index: %w", err)
+	}
+
+	indexManifest, err := index.IndexManifest()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read index manifest: %w", err)
+	}
+
+	if len(indexManifest.Manifests) == 0 {
+		return nil, fmt.Errorf("no manifests found in OCI layout")
+	}
+
+	// Select the manifest based on digest selector
+	var manifest v1.Descriptor
+	if digestSelector != "" {
+		// Find manifest with matching digest
+		found := false
+		for _, m := range indexManifest.Manifests {
+			if m.Digest.String() == digestSelector {
+				manifest = m
+				found = true
+				break
+			}
+		}
+		if !found {
+			return nil, fmt.Errorf("digest %s not found in OCI layout", digestSelector)
+		}
+		i.logger.Debug().Str("digest", digestSelector).Msg("selected image by digest")
+	} else {
+		// If multiple images exist, require explicit selection to avoid mistakes
+		if len(indexManifest.Manifests) > 1 {
```

Comment:
This might be refactored more as a guard instead of as if/else, meaning checking the condition at the top, potentially setting the first manifest by default, etc.